runtime/test_artifacts/generator.go

```diff
@@ -6,108 +6,149 @@ import (
-	fish, fishPrivateKeys := NewActors(coreTypes.ActorType_ACTOR_TYPE_FISH, numFisherman)
+type GenesisOption func(*genesis.GenesisState)
+
+// IMPROVE: Extend the utilities here into a proper genesis suite in the future.
+func NewGenesisState(numValidators, numServicers, numApplications, numFisherman int, genesisOpts ...GenesisOption) (genesisState *genesis.GenesisState, validatorPrivateKeys []string) {
+	applications, appPrivateKeys := NewActors(coreTypes.ActorType_ACTOR_TYPE_APP, numApplications, DefaultChains)
+	validators, validatorPrivateKeys := NewActors(coreTypes.ActorType_ACTOR_TYPE_VAL, numValidators, nil)
+	servicers, servicerPrivateKeys := NewActors(coreTypes.ActorType_ACTOR_TYPE_SERVICER, numServicers, DefaultChains)
+	fishermen, fishPrivateKeys := NewActors(coreTypes.ActorType_ACTOR_TYPE_FISH, numFisherman, DefaultChains)
+
+	allActorsKeys := append(append(append(validatorPrivateKeys, servicerPrivateKeys...), fishPrivateKeys...), appPrivateKeys...)
+	allActorAccounts := newAccountsWithKeys(allActorsKeys)
 
 	genesisState = &genesis.GenesisState{
 		GenesisTime:   timestamppb.Now(),
 		ChainId:       DefaultChainID,
 		MaxBlockBytes: DefaultMaxBlockBytes,
 		Pools:         NewPools(),
-		Accounts:      NewAccounts(numValidators+numServicers+numApplications+numFisherman, append(append(append(validatorPrivateKeys, snPrivateKeys...), fishPrivateKeys...), appsPrivateKeys...)...), // TODO(olshansky): clean this up
-		Applications:  apps,
-		Validators:    vals,
+		Accounts:      allActorAccounts,
+		Applications:  applications,
+		Validators:    validators,
 		Servicers:     servicers,
-		Fishermen:     fish,
+		Fishermen:     fishermen,
 		Params:        DefaultParams(),
 	}
 
-	// TODO: Generalize this to all actors and not just validators
+	for _, o := range genesisOpts {
+		o(genesisState)
+	}
+
+	// TECHDEBT: Generalize this to all actors and not just validators
 	return genesisState, validatorPrivateKeys
 }
 
+func WithActors(actors []*coreTypes.Actor, actorKeys []string) func(*genesis.GenesisState) {
+	return func(genesis *genesis.GenesisState) {
+		newActorAccounts := newAccountsWithKeys(actorKeys)
+		genesis.Accounts = append(genesis.Accounts, newActorAccounts...)
+		for _, actor := range actors {
+			switch actor.ActorType {
+			case types.ActorType_ACTOR_TYPE_APP:
+				genesis.Applications = append(genesis.Applications, actor)
+			case coreTypes.ActorType_ACTOR_TYPE_VAL:
+				genesis.Validators = append(genesis.Validators, actor)
+			case coreTypes.ActorType_ACTOR_TYPE_SERVICER:
+				genesis.Servicers = append(genesis.Servicers, actor)
+			case coreTypes.ActorType_ACTOR_TYPE_FISH:
+				genesis.Fishermen = append(genesis.Fishermen, actor)
+			default:
+				panic(fmt.Sprintf("invalid actor type: %s", actor.ActorType))
+			}
+		}
+	}
+}
+
 func NewDefaultConfigs(privateKeys []string) (cfgs []*configs.Config) {
 	for i, pk := range privateKeys {
+		postgresSchema := "node" + strconv.Itoa(i+1)
```

Comment:
1. Where is this convention derived from? Can we link back to it or some docs in a comment?
2. nit: I think it would improve maintainability slightly if we move this to a helper function.

Reply:
It is very outside of the scope of this PR so I'm going to leave a TECHDEBT comment:

```
// TECHDEBT: This is used for the `node_schema` field in `PersistenceConfig` and enables
// different nodes sharing the same database while being isolated from each other.
// The naming convention should be changed to be more reflective of the node (e.g. <actor_type>_<address>),
// which would require all related tooling and documentation to be updated as well.
```